templates/Makefile.erb

```diff
@@ -79,8 +79,11 @@ cleanup-components: <%= @components.map {|comp| "#{comp.name}-cleanup" }.join("
 	touch <%= comp.name %>-patch
 
 <%= comp.name %>-configure: <%= comp.name %>-patch <%= list_component_dependencies(comp).join(" ") %>
+	<%- if comp.get_build_dir -%>
+	[ -d <%= comp.get_build_dir %> ] || mkdir <%= comp.get_build_dir %>
```

Comment:
Testing for existence isn't strictly necessary here since mkdir will still exit successfully if the directory already exists. (Nitpick, not blocking).